org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/BuilderTests.java

```diff
@@ -650,7 +647,7 @@ protected static void expectCompileProblem(IPath project, String expectedProblem
 
 		List<String> expectedProblemMessages = Arrays.asList(expectedProblemMessage);
 		assertEquals("expected compile problem not observed",
-				expectedProblemMessages, actualProblemMessages);
+				expectedProblemMessages.toString(), actualProblemMessages.toString());
```

Comment:
Curious - what makes this change necessary ??

Reply:
I stumbled upon that, too, and figured that perhaps the JUnit view gives better support for comparing right and left if both are already strings?

Reply:
I had to analyse related test failure I had to run that in IDE few times and the toString() produces clear test fail message with a readable diff, the previous version just failed with assert